packages/chain-agnostic-permission/src/adapters/caip-permission-adapter-permittedChains.ts

```diff
@@ -145,3 +145,92 @@ export const setPermittedEthChainIds = (
 
   return updatedCaveatValue;
 };
+
+/**
+ * Filters the scopes object to only include:
+ * - Scopes without references (e.g. "wallet:")
+ * - CAIP-2 ChainId scopes for the given chainIDs
+ *
+ * @param scopesObject - The scopes object to filter.
+ * @param chainIds - The CAIP-2 chainIDs to filter for.
+ * @returns The filtered scopes object.
+ */
+const filterChainScopesObjectByChainId = (
+  scopesObject: InternalScopesObject,
+  chainIds: CaipChainId[],
+): InternalScopesObject => {
+  const updatedScopesObject: InternalScopesObject = {};
+
+  Object.entries(scopesObject).forEach(([key, scopeObject]) => {
+    // Cast needed because index type is returned as `string` by `Object.entries`
+    const scopeString = key as keyof typeof scopesObject;
+    // If its a wallet scope or a wallet:* scope we don't filter it
+    if (isWalletScope(scopeString)) {
+      updatedScopesObject[scopeString] = scopeObject;
+    } else if (chainIds.includes(scopeString)) {
```

Comment:
can these two statements be combined into one guard?

Reply:
Done here: https://github.com/MetaMask/core/pull/5536/commits/305d18ddcbd965ee591a88ea2b21c4250bdd714e